### PR TITLE
[mlir][spirv] Remove `enableFastMathMode` flag from SPIR-V conversion

### PR DESCRIPTION
**Pull request by Hardcode84**
Most of arith/math ops support fastmath attribute, use it instead of global flag.

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
Makes sense to me

File: mlir/test/Conversion/ArithToSPIRV/fast-math.mlir [hunk 1]
```diff
@@ -8,7 +8,7 @@ module attributes {
 // CHECK-SAME: %[[LHS:.+]]: f32, %[[RHS:.+]]: f32
 func.func @cmpf_ordered(%arg0 : f32, %arg1 : f32) -> i1 {
   // CHECK: %[[T:.+]] = spirv.Constant true
-  %0 = arith.cmpf ord, %arg0, %arg1 : f32
+  %0 = arith.cmpf ord, %arg0, %arg1 fastmath<fast> : f32
```
Comment (ScottTodd) [inline mlir/test/Conversion/ArithToSPIRV/fast-math.mlir:11]: Is there a pass in upstream MLIR to add fastmath attributes to `arith` dialect ops?

We were using the spirv fast math mode downstream here: https://github.com/openxla/iree/blob/aacdd33eb57f79711ecae088dbb37c9bd17d7031/compiler/plugins/target/WebGPUSPIRV/WebGPUSPIRVTarget.cpp#L148-L159 . To adapt to this change, I think we'd want to add these attributes to all arith ops prior to running arith to spirv patterns here: https://github.com/openxla/iree/blob/aacdd33eb57f79711ecae088dbb37c9bd17d7031/compiler/src/iree/compiler/Codegen/SPIRV/ConvertToSPIRVPass.cpp#L655-L660.

So in our case, we start with modules that don't generally use fastmath, but then partway through our lowering process we specialize for one target that requires it. For that, we want this same "just use fast math" behavior, without needing to enumerate all possible ops that need an attribute ourselves.

Comment (kuhar) [reply mlir/test/Conversion/ArithToSPIRV/fast-math.mlir:11]: fastify-math pass sounds like a good idea

Comment (Hardcode84) [reply mlir/test/Conversion/ArithToSPIRV/fast-math.mlir:11]: > Is there a pass in upstream MLIR to add fastmath attributes to arith dialect ops?

I don't think upstream have such pass, but it should be straightforward to implement such pass without enumerating specific ops. All arith ops which support fastmath are implementing `ArithFastMathInterface`, so pass may look something like this:
```
void runOnOperation() {
    getOperation()->walk([](ArithFastMathInterface iface) {
        ... set fastmath attribute
    });
}
```

**Pull request merged**